### DISCUSSION TITLE
K1 Max mainboard:  Does it have enough resources to run mainsail aloing with all the good stuff?

### DISCUSSION THREAD

**Discussion by martindellavecchia** [Guilouz/Creality-Helper-Script-Wiki#524] [General]
Considering that the K1/K1 max came with a basic web interface and limited functionality, it makes me think that creality cheaped out in the mainboard, and that the K1/K1 max comes with a not sunch a powerful board. Following that rationale my question would be: could installing mainsail on my k1 max may cause any resource (cpu/memory) shortage that may cause any issue ?

**Comment by TheWebMachine** [comment]
If you are worried about print issues, generally speaking, no. The klipper side and the actual printer side of things are separate, with their own discreet MCUs communicating over a serial connection.

Could you, in theory, max the CPU and slow the Klipper MCU down until it no longer sends timely gcode to the printer MCU? Yes, but I've yet to see it happen.

Creality "cheapened" the web interface to protect novice users from themselves. That's why there was such outcry from the maker community (to which Creality caved) to release a readily-rooted firmware moving forward. 

Unless you rely on Creality Cloud or the Creality App for printing, you can safely "remove" [disable] the Creality Web interface all together and let Fluidd or Mainsail take over port 80 as the sole/main web interface. That way you won't have Creality's version of things using up resources to alleviate any concerns on that front.

**Comment by martindellavecchia** [comment]
Cool, thanks it's now clear to me, looks like it's safe performance wise. Is there any drawback I should be aware, i.e. lidar support, stop reciving official firmware updates, etc?

**Reply by TheWebMachine** [reply]
Nope. Besides Creality Cloud/App access, everything else should work as intended. Firmware updates always require a factory reset and set back up, but there are ways to backup your config when the time comes.

**Reply by martindellavecchia** [reply]
Super!, really appreciated.
